### PR TITLE
Add test case for JSON marshaling

### PR DESCRIPTION
**Pull request by rahulkarmore**
Hi creator,
Added test cases for JSON Resources :
ExternalGroup
Helps: https://github.com/google/go-github/issues/55 

I have followed all the guidelines to contribute to this repository.
Please revert in case of any changes are needed.
I am a first-time contributor to Hackoctober, It will be a great help if u can encourage me & accept my contribution.

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @rahulkarmore !
LGTM.
Merging.

**Comment by gmlewis** [comment]
Whups, I need to wait for the CLA to be signed and resolved before I can merge this PR.

**Comment by rahulkarmore** [comment]
Close this merge request, New merge request will be 
https://github.com/google/go-github/pull/2541

**Pull request closed without merging**
